src/recommendpolicy/recommendPolicy.go

```diff
@@ -180,3 +176,43 @@ func uniqueNsDeploy(deployName, deployNamespace string) *types.Deployment {
 
 	return &deploy
 }
+
+func GetHardenPolicy(deployments *v1.DeploymentList, replicaSets *v1.ReplicaSetList, statefulSets *v1.StatefulSetList, nsNotFilter []string) []types.KnoxSystemPolicy {
+
+	var policies []types.KnoxSystemPolicy
+	if !isLatest() {
+		if _, err := DownloadAndUnzipRelease(); err != nil {
+			log.Error().Msgf("Unable to download %v", err.Error())
+		}
+	}
+	for _, d := range deployments.Items {
+		deploy := uniqueNsDeploy(d.Name, d.Namespace)
+
+		if deploy != nil {
+			DeployNsName = append(DeployNsName, *deploy)
+		}
+
+		for _, ns := range nsNotFilter {
+			if d.Namespace != ns {
+				policies = append(policies, generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)...)
+			}
+		}
+	}
+
+	for _, r := range replicaSets.Items {
+		for _, ns := range nsNotFilter {
+			if r.Namespace != ns && len(r.ObjectMeta.OwnerReferences) == 0 {
```

Comment:
Why is the check `len(r.ObjectMeta.OwnerReferences) == 0` not present for statefulsets/deployments?

Reply:
It can be possible that deployment are managed by `job` owner. I have added the checks for those cases.